plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/index/HudiColumnStatsIndexSupport.java

```diff
@@ -56,41 +65,54 @@ public class HudiColumnStatsIndexSupport
+    protected final TupleDomain<String> regularColumnPredicates;
+    private final List<String> regularColumns;
+    private final Duration columnStatsWaitTimeout;
 
-    public HudiColumnStatsIndexSupport(HoodieTableMetaClient metaClient)
+    public HudiColumnStatsIndexSupport(ConnectorSession session, HoodieTableMetaClient metaClient, HoodieTableMetadata metadataTable, TupleDomain<HiveColumnHandle> regularColumnPredicates)
     {
-        super(log, metaClient);
+        this(log, session, metaClient, metadataTable, regularColumnPredicates);
     }
 
-    public HudiColumnStatsIndexSupport(Logger log, HoodieTableMetaClient metaClient)
+    public HudiColumnStatsIndexSupport(Logger log, ConnectorSession session, HoodieTableMetaClient metaClient, HoodieTableMetadata metadataTable, TupleDomain<HiveColumnHandle> regularColumnPredicates)
     {
         super(log, metaClient);
+        this.columnStatsWaitTimeout = getColumnStatusWaitTimeout(session);
+        this.regularColumnPredicates = regularColumnPredicates.transformKeys(HiveColumnHandle::getName);
+        this.regularColumns = this.regularColumnPredicates
+                .getDomains().get().entrySet().stream().map(Map.Entry::getKey).collect(Collectors.toList());
+        if (regularColumnPredicates.isAll() || !regularColumnPredicates.getDomains().isPresent()) {
+            this.statsByFileNameFuture = CompletableFuture.completedFuture(Optional.empty());
+        }
+        else {
+            // Get filter columns
+            List<String> encodedTargetColumnNames = regularColumns
+                    .stream()
+                    .map(col -> new ColumnIndexID(col).asBase64EncodedString()).collect(Collectors.toList());
+            statsByFileNameFuture = CompletableFuture.supplyAsync(() ->
+                    Optional.of(metadataTable.getRecordsByKeyPrefixes(encodedTargetColumnNames,
+                                    HoodieTableMetadataUtil.PARTITION_NAME_COLUMN_STATS, true)
+                            .collectAsList()
+                            .stream()
+                            .filter(f -> f.getData().getColumnStatMetadata().isPresent())
+                            .map(f -> f.getData().getColumnStatMetadata().get())
+                            .collect(Collectors.groupingBy(
+                                    HoodieMetadataColumnStats::getFileName,
+                                    Collectors.toMap(
+                                            HoodieMetadataColumnStats::getColumnName,
+                                            Function.identity())))));
+        }
     }
 
     @Override
     public Map<String, List<FileSlice>> lookupCandidateFilesInMetadataTable(
-            HoodieTableMetadata metadataTable, Map<String, List<FileSlice>> inputFileSlices,
-            TupleDomain<String> regularColumnPredicates)
+            Map<String, List<FileSlice>> inputFileSlices, TupleDomain<String> regularColumnPredicates)
```

Comment:
Looks like `#lookupCandidateFilesInMetadataTable` API is no longer used.  Should it be removed?